incubator/hnc/pkg/object/object.go

```diff
@@ -35,7 +36,9 @@ func Canonical(inst *unstructured.Unstructured) *unstructured.Unstructured {
 	// Non-HNC labels:
 	newLabels := map[string]string{}
 	for k, v := range inst.GetLabels() {
-		if !strings.HasPrefix(k, api.MetaGroup) {
+		if !strings.HasPrefix(k, api.MetaGroup) &&
```

Comment:
This will break if we add any other prefixes like `tree`. How about something like:

```
if !strings.HasSuffix(metaPrefix(k), api.MetaGroup) { ... }

...

func metaPrefix(s string) string {
  p := strings.Split(s, "/")
  if len(p) == 1 {
    return ""
  }
  return p[0]
}
```

Reply:
Reference: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set